src/plugins/data/common/es_query/es_query/build_es_query.ts

```diff
@@ -41,7 +41,7 @@ export interface EsQueryConfig {
  * config contains dateformat:tz
  */
 export function buildEsQuery(
-  indexPattern: IIndexPattern | null,
+  indexPattern: IIndexPattern | undefined,
```

Comment:
Why was this changed? Does it make sense to call it with no index pattern?
And if this is the interface, why not `indexPattern?: IIndexPattern`?

Reply:
cannot do it cause optional parameters are defined at the end of the parameter list, after any required parameters. See:

```
export function buildEsQuery(
  indexPattern: IIndexPattern | undefined,  // <- required parameter
  queries: Query | Query[],// <- required parameter
  filters: Filter | Filter[], // <- required parameter
  config: EsQueryConfig = { // <- required parameter
    allowLeadingWildcards: false,
    queryStringOptions: {},
    ignoreFilterIfFieldNotInIndex: false,
  }
```